#13108: System event OnResourceSort to provide information whether re…

SiarAneas opened the pull request:
### What does it do?

Enhanced modResourceSortProcessor to provide sorted resource ID (the one that was dragged&dropped in resource tree) and an indication whether the sorted resources' parent got changed during that operation through fireAfterSort() invoking the 'OnResourceSort' event
### Why is it needed?

See #13108 
### Related issue(s)/PR(s)

Resolves #13108

Commit:
#13108: System event OnResourceSort to provide information whether resource in tree was moved under new parent

rtripault commented on core/model/modx/processors/resource/sort.class.php:
```diff
@@ -39,6 +40,8 @@ public function process() {
         $this->getNodesFormatted($data, $this->getProperty('parent', 0));
 
         $this->fireBeforeSort();
+        $resourceBeforeSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
```
I believe this processor is also handling context sorting, so `source_pk` could be a `modContext.key`

SiarAneas replied:
Just realized, that's true - bit confusing name for the processor then... I will provide a check whether $this->getProperty('source_pk') is a valid resource by checking whether $this->getProperty('source_type')==="modContext"

Commit:
#13108: Added check to ensure moved resource is of type modResource

Commit:
13108: Added check to ensure moved resource is of type modResource

Author:    Andreas Mauter <andreas.mauter@passion-works.net>
Date:      Wed Aug 31 09:46:03 2016 +0200

Commit:
Merge branch 'issue-13108' of github.com:SiarAneas/revolution into issue-13108

Mark-H commented on core/model/modx/processors/resource/sort.class.php:
```diff
@@ -39,6 +40,10 @@ public function process() {
         $this->fireBeforeSort();
+        if($this->getProperty('source_type')==="modResource"){
+            $resourceBeforeSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
+            $oldParentID = $resourceBeforeSort->get('parent');
```
This could use a check if $resourceBeforeSort has a valid resource, otherwise this could cause a fatal error "method get on null" if for whatever reason the resource can't be loaded.

Mark-H commented on core/model/modx/processors/resource/sort.class.php:
```diff
@@ -62,6 +67,12 @@ public function process() {
+        if($this->getProperty('source_type')==="modResource"){
+            $resourceAfterSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
+            $newParentID = $resourceAfterSort->get('parent');
+            if($oldParentID !== $newParentID) $this->resourceMovedUnderNewParent = true;
```
The body of the if clause should be wrapped in curly brackets `{ }` and on its own line

Mark-H commented on core/model/modx/processors/resource/sort.class.php:
```diff
@@ -62,6 +67,12 @@ public function process() {
 
+        if($this->getProperty('source_type')==="modResource"){
+            $resourceAfterSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
+            $newParentID = $resourceAfterSort->get('parent');
```
Same as above, if for some weird reason the resource can't be loaded, this would cause a fatal error "method get on null".

Mark-H commented:
Pending the tweaks to account for potential fatal errors if something goes wrong, I think this is a nice improvement.

sdrenth commented:
@SiarAneas Are you planning on fixing the requested changes? Would be nice so we can get this PR merged ;)

Mark-H commented:
Closing as there has been no contributor feedback and I'm not comfortable merging it without those checks. If anyone else wants to recreate the PR with the requested fixes, be my guest :)
